bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java

```diff
@@ -60,4 +62,32 @@ public void testAccessTokenProvider() throws Exception {
     assertThat(table).isNotNull();
     assertThat(accessTokenProvider.getCallCount()).isEqualTo(2);
   }
+
+  @Test
+  public void testAccessTokenProvider_withConfig() throws IOException {
+    BigQueryCredentialsSupplier credentialsSupplier =
+        new BigQueryCredentialsSupplier(
+            Optional.of(DefaultCredentialsDelegateAccessTokenProvider.class.getCanonicalName()),
+            Optional.of("some-static-token"),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+    Credentials credentials = credentialsSupplier.getCredentials();
+    assertThat(credentials).isInstanceOf(AccessTokenProviderCredentials.class);
+    DefaultCredentialsDelegateAccessTokenProvider accessTokenProvider =
+        (DefaultCredentialsDelegateAccessTokenProvider)
+            ((AccessTokenProviderCredentials) credentials).getAccessTokenProvider();
+    assertThat(accessTokenProvider.getCallCount()).isEqualTo(0);
+    assertThat(accessTokenProvider.getAccessToken().getTokenValue()).isEqualTo("some-static-token");
```

Comment:
The test is not giving a valid access token, but a dummy string. This will cause the call to the BigQuery API to fail. I suggest that the field name to be "config" and to check that its value was properly set. The `getAccessToken()` implementation can remain the same.